### ISSUE TITLE
Will Caffe support TX1's FP16?

### ISSUE THREAD

**Issue by leejiajun** [BVLC/caffe#3582]
Will Caffe support TX1's FP16?
Have any suggestions about that?

Thanks!

**Comment by Darwin2011** [comment]
NVIDIA's Caffe fork has one branch which seems to support FP16.

https://github.com/NVIDIA/caffe/tree/experimental/fp16

**Comment by leejiajun** [comment]
@Darwin2011 thank you. But it seems not to support.

**Comment by leejiajun** [comment]
@Darwin2011 because I have not found “#include <cuda_fp16.h>” and “half” in the whole project.
Do you have any other tips？

**Comment by thatguymike** [comment]
That branch does support fp16.  native fp16 only works currently on jetson tx1 and is designed for inference (forward passes).  fp16 is handled via the templates for mtype and dtype.  CAFFE_FP16 and float16 are the things you are looking for.  include/caffe/util has the headers for fp16 handling.

**Comment by leejiajun** [comment]
@Darwin2011 @thatguymike 
That branch does support fp16, thank you.

**Comment by feiliz** [comment]
Are there somebody know  that whether mali T880  support Fp16?

**Comment by leejiajun** [comment]
@Darwin2011 
@thatguymike 
I got an error when I am using _caffe-experimental-fp16_'s pycaffe. https://github.com/NVIDIA/caffe/issues/117

Could you give me some clues?

**Comment by psyhtest** [comment]
@feiliz All generations of the Mali Midgard architecture (including the 4th generations Mali-T880) support FP16.

**Comment by loulansuiye** [comment]
how  this issue now?

**Comment by naibaf7** [comment]
Working on it (FP16 support on CUDA and OpenCL Caffe). Expect a release by mid January at latest.

**Comment by jklontz** [comment]
Any update on this? Thanks in advance.

**Comment by naibaf7** [comment]
Yes, sorry, it is a bit delayed, but basically FP16 works: https://github.com/naibaf7/caffe
